src/SBSpergel.cpp

```diff
@@ -0,0 +1,715 @@
+        double _target;
+    };
+
+    double SpergelInfo::calculateFluxRadius(const double& flux_frac) const
+    {
+        // Calcute r such that L(r/r0) / L_tot == flux_frac
+
+        // These seem to bracket pretty much every reasonable possibility
+        // that I checked in Mathematica, though I'm not very happy about
+        // the lower bound....
+        double z1=1.e-16;
+        double z2=20.0;
+        SpergelIntegratedFlux func(_nu, _gamma_nup2, flux_frac);
+        Solve<SpergelIntegratedFlux> solver(func, z1, z2);
+        solver.setMethod(Brent);
+        solver.bracketLowerWithLimit(0.0); // Just in case...
+        double R = solver.root();
+        dbg<<"flux_frac = "<<flux_frac<<std::endl;
+        dbg<<"r/r0 = "<<R<<std::endl;
+        return R;
+    }
+
+    double SpergelInfo::stepK() const
+    {
+        if (_stepk == 0.) {
+            double R = calculateFluxRadius(1.0 - _gsparams->folding_threshold);
+            if (_truncated && _trunc < R)  R = _trunc;
+            // Go to at least 5*re
+            R = std::max(R,_gsparams->stepk_minimum_hlr);
+            dbg<<"R => "<<R<<std::endl;
+            _stepk = M_PI / R;
+            dbg<<"stepk = "<<_stepk<<std::endl;
+        }
+        return _stepk;
+    }
+
+    double SpergelInfo::maxK() const
+    {
+        if(_maxk == 0.) {
+            if (_truncated) buildFT();
+            else {
+                // Solving (1+k^2)^(-1-nu) = maxk_threshold for k
+                // exact:
+                //_maxk = std::sqrt(std::pow(gsparams->maxk_threshold, -1./(1+_nu))-1.0);
+                // approximate 1+k^2 ~ k^2 => good enough:
+                _maxk = std::pow(_gsparams->maxk_threshold, -1./(2*(1+_nu)));
+            }
+        }
+        return _maxk;
+    }
+
+    double SpergelInfo::getHLR() const
+    {
+        if (_re == 0.0) calculateHLR();
+        return _re;
+    }
+
+    double SpergelInfo::getFluxFraction() const
+    {
+        if (_flux == 0.) {
+            // Calculate the flux of a truncated profile (relative to the integral for
+            // an untruncated profile).
+            if (_truncated) {
+                SpergelIntegratedFlux func(_nu, _gamma_nup2, 0.0);
+                _flux = func(_trunc);
+                dbg << "Flux fraction = " << _flux << std::endl;
+            } else {
+                _flux = 1.;
+            }
+        }
+        return _flux;
+    }
+
+    void SpergelInfo::calculateHLR() const
+    {
+        dbg<<"Find HLR for (nu,trunc) = ("<<_nu<<","<<_trunc<<")"<<std::endl;
+        SpergelIntegratedFlux func(_nu, _gamma_nup2, 0.5*getFluxFraction());
+        double b1 = 0.001; // These are pretty conservative...
+        double b2 = 25.0;
+        Solve<SpergelIntegratedFlux> solver(func, b1, b2);
+        xdbg<<"Initial range is "<<b1<<" .. "<<b2<<std::endl;
+        solver.setMethod(Brent);
+        solver.bracketLowerWithLimit(0.); // Expand if needed
+        solver.bracketUpper();
```

Comment:
The range you have is actually more than sufficient.  So I don't think you need these bracket calls.

I believe the range of _re values for nu in (-0.85, 0.85) is (0.13, 1.92).  So (0.001, 25) is completely sufficient without any expansion.  You could even shrink it a bit if you want to make this calculation more efficient, but that's not really all that necessary I suspect.  The solver probably zeros in on the right answer pretty quickly as it is.